marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -315,6 +301,86 @@ function getPrimaryValue(entityInstance, entityDefinition) {
   }
 }
 
+//Helper function to add properties to each result instance under results array in searchReponse
+function addEntitySpecificProperties(result, entityInfo, selectedPropertyMetadata) {
+  let instance = null;
+  let entityTitle = entityInfo.entityName;
+  result.entityProperties = [];
+
+  try {
+    instance = getEntityInstance(result.uri);
+  } catch (error) {
+    console.log(`Unable to obtain entity instance from document with URI '${result.uri}'; will not add entity properties to its search result`);
+  }
+
+  if (instance != null) {
+    let entityDef = entityInfo.entityModel.definitions[entityTitle];
+    const entityInstance = instance[entityTitle];
+    if (!entityInstance) {
+      console.log(`Unable to obtain entity instance from document with URI '${result.uri}' and entity name '${entityTitle}'; will not add entity properties to its search result`);
+    } else {
+      selectedPropertyMetadata.forEach(parentProperty => {
+        result.entityProperties.push(getPropertyValues(parentProperty, entityInstance));
+      });
+
+      addPrimaryKeyToResult(result, entityInstance, entityDef);
+    }
+  }
+}
+
+function addGenericEntityProperties(result) {
+  let instance = null;
+  let entityTitle = "";
+  let createdOnDate = "";
+  let entityModel = {};
+  result.primaryKey = {};
+  result.identifier = {};
+  result.entityName = "";
+  result.createdOn = "";
+
+  try {
+    instance = getEntityInstance(result.uri);
+  } catch (error) {
+    console.log(`Unable to obtain entity instance from document with URI '${result.uri}'; will not add entity properties to its search result`);
+  }
+
+  if (instance != null) {
+    entityTitle = instance.hasOwnProperty("info") ? instance.info.title : Object.keys(instance)[0];
+    entityModel = entityLib.findModelByEntityName(entityTitle);
+    let entityDef = entityModel.definitions[entityTitle];
+    const entityInstance = instance[entityTitle];
+    if (!entityInstance) {
+      console.log(`Unable to obtain entity instance from document with URI '${result.uri}' and entity name '${entityTitle}'; will not add entity properties to its search result`);
+    } else {
+        addPrimaryKeyToResult(result, entityInstance, entityDef);
+    }
+  }
+  try {
```

Comment:
If there is no instance, then is no point in returning the identifier, entityName and createdOn

probably we should check this first in the beginning of the function and then if the condition fails, we should process the identifier, createdOn and other details and return the object 

```
if(! instance) {
  return;
}
```

@rjrudin Can you confirm this

Reply:
@rahulvudutala @rjrudin - Well we can restrict identifier and entityName but createdOn doesn't seem to depend on whether instance is available in the doc or not. It just captures the datahubCreatedOn date from document metadata.
Any thoughts ?

Reply:
I think identifier and createdOn are all still relevant even if we can't find the instance in the document. This is assuming that the URI would be the identifier and createdOn comes from metadata.

Reply:
@rahulvudutala - Then let us not make any changes in this logic. We can update it later if required.